MvvmCross/Core/MvxSetupSingleton.cs

```diff
@@ -169,17 +174,32 @@ protected override void Dispose(bool isDisposing)
             base.Dispose(isDisposing);
         }
 
+        private Exception SetupException { get; set; }
         private void StartSetupInitialization()
         {
             IsInitialisedTaskCompletionSource = new TaskCompletionSource<bool>();
             _setup.InitializePrimary();
             Task.Run(async () =>
             {
-                _setup.InitializeSecondary();
+                try
+                {
+                    _setup.InitializeSecondary();
+                }
+                catch(Exception ex)
+                {
+                    SetupException = ex;
```

Comment:
If this exception would be rethrown at a later point, it could be worth using `ExceptionDispatchInfo.Capture(ex)` to give a more meaningful stack trace.